src/github.js

```diff
@@ -36,72 +37,70 @@ async function createStatusCheck(repoToken, reportData, markupData, conclusion)
   });
 
   if (commentsResponse.status !== 200) {
-    core.setFailed(`Failed to list PR comments. Error code: ${commentsResponse.status}`);
-    return;
+    core.info(`Failed to list PR comments. Error code: ${commentsResponse.status}.  Will create new comment instead.`);
+    return null;
   }
 
-  const prefixedMarkupData = '<!-- im-open/process-dotnet-test-results -->\n' + markupData;
-
-  const existingComment = commentsResponse.data.find(comment => comment.body.startsWith(prefixedMarkupData));
-
-  if (existingComment === undefined) {
-    await createComment(octokit, prefixedMarkupData);
-  } else {
-    core.info(`Updating PR Comment...`);
-    const response = await octokit.rest.issues.updateComment({
-      comment_id: existingComment.id,
-      owner: github.context.repo.owner,
-      repo: github.context.repo.repo,
-      body: prefixedMarkupData
-    });
-
-    if (response.status !== 200) {
-      core.setFailed(`Failed to update PR comment. Error code: ${response.status}`);
-    } else {
-      core.info(`Updated PR comment: ${response.data.id} with response status ${response.status}`);
-    }
+  const existingComment = commentsResponse.data.find(c => c.body.startsWith(markupPrefix));
+  if (!existingComment) {
+    core.info('An existing dotnet test results comment was not found, create a new one instead.');
   }
+  return existingComment ? existingComment.id : null;
 }
 
 async function createPrComment(repoToken, markupData, updateCommentIfOneExists) {
   try {
     if (github.context.eventName != 'pull_request') {
-      core.info('This event was not triggered by a pull_request.  No comment will be created.');
+      core.info('This event was not triggered by a pull_request.  No comment will be created or updated.');
       return;
     }
 
     const octokit = github.getOctokit(repoToken);
+
+    let existingCommentId = null;
     if (updateCommentIfOneExists) {
-      await createOrUpdateComment(octokit, markupData);
+      core.info('Checking for existing comment on PR....');
+      existingCommentId = await lookForExistingComment(octokit);
+    }
+    ``;
+    let response;
+    let success;
+    if (existingCommentId) {
+      core.info(`Updating existing PR #${existingCommentId} comment...`);
+      response = await octokit.rest.issues.updateComment({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        body: `${markupPrefix}\n${markupData}`,
+        comment_id: existingCommentId
+      });
+      success = response.status === 200;
+    } else {
+      core.info(`Creating a new PR comment...`);
+      response = await octokit.rest.issues.createComment({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        body: `${markupPrefix}\n${markupData}`,
+        issue_number: github.context.payload.pull_request.number
+      });
+      success = response.status === 201;
+    }
+
+    let action = existingCommentId ? 'create' : 'update';
```

Comment:
The `action` isn't getting set after this so `const` would work. Otherwise you could declare and set `action` like `response` and `success` above and remove this ternary. Or you could leave it alone because it's not a huge deal and it works 😆